### PR TITLE
Develop tests for haproxy helper

### LINKED ISSUE #20940: Add unit tests for the new `HAProxy helper`
| Epic | 
|--|
|https://github.com/wazuh/wazuh/issues/20887|

## Description
Add unit tests for all the `HAProxy helper` code. The tests should not include new warnings.

### PR DESCRIPTION
**Pull request by nico-stefani**
|Related issue|
|---|
| #20940 |

## Description

This PR closes #20940. Adds unit tests for the HAProxy Helper functionality.

## Tests

All the tests are passed without any warning.

```python
(framework) ➜  wazuh git:(20940-develop-tests-for-haproxy-helper) ✗ PYTHONPATH=$WAZUH_REPO/api:$WAZUH_REPO/framework python -m pytest framework/
============================= test session starts ==============================
platform linux -- Python 3.10.13, pytest-7.3.1, pluggy-1.4.0
rootdir: /home/nstefani/git/wazuh/framework
configfile: pytest.ini
plugins: aiohttp-1.0.4, sugar-1.0.0, cov-5.0.0, asyncio-0.18.1, tavern-1.23.5, trio-0.8.0, html-2.1.1, metadata-3.1.1, anyio-4.3.0
asyncio: mode=auto
collected 2373 items

framework/scripts/tests/test_agent_groups.py ..............              [  0%]
framework/scripts/tests/test_agent_upgrade.py ...............            [  1%]
framework/scripts/tests/test_cluster_control.py ......                   [  1%]
framework/scripts/tests/test_rbac_control.py .........                   [  1%]
framework/scripts/tests/test_wazuh_clusterd.py ........                  [  2%]
framework/scripts/tests/test_wazuh_logtest.py ......................     [  3%]
framework/wazuh/core/cluster/dapi/tests/test_dapi.py ................... [  3%]
.............                                                            [  4%]
framework/wazuh/core/cluster/hap_helper/tests/test_hap_helper.py ....... [  4%]
.......................................                                  [  6%]
framework/wazuh/core/cluster/hap_helper/tests/test_proxy.py ............ [  6%]
........................................................................ [  9%]
...............                                                          [ 10%]
framework/wazuh/core/cluster/hap_helper/tests/test_wazuh.py ..........   [ 10%]
framework/wazuh/core/cluster/tests/test_client.py ................       [ 11%]
framework/wazuh/core/cluster/tests/test_cluster.py ..................... [ 12%]
................                                                         [ 13%]
framework/wazuh/core/cluster/tests/test_common.py ...................... [ 14%]
..............................................................           [ 16%]
framework/wazuh/core/cluster/tests/test_control.py ......                [ 17%]
framework/wazuh/core/cluster/tests/test_local_client.py ..............   [ 17%]
framework/wazuh/core/cluster/tests/test_local_server.py ................ [ 18%]
........                                                                 [ 18%]
framework/wazuh/core/cluster/tests/test_master.py ...................... [ 19%]
.........................                                                [ 20%]
framework/wazuh/core/cluster/tests/test_server.py ...................... [ 21%]
.......                                                                  [ 21%]
framework/wazuh/core/cluster/tests/test_utils.py ....................    [ 22%]
framework/wazuh/core/cluster/tests/test_worker.py ...................... [ 23%]
............                                                             [ 24%]
framework/wazuh/core/tests/test_active_response.py ....................  [ 24%]
framework/wazuh/core/tests/test_agent.py ............................... [ 26%]
........................................................................ [ 29%]
..............................................                           [ 31%]
framework/wazuh/core/tests/test_cdb_list.py ............................ [ 32%]
..........                                                               [ 32%]
framework/wazuh/core/tests/test_common.py .........                      [ 33%]
framework/wazuh/core/tests/test_configuration.py ....................... [ 34%]
..................................................................       [ 36%]
framework/wazuh/core/tests/test_decoder.py ................              [ 37%]
framework/wazuh/core/tests/test_exception.py ..........                  [ 38%]
framework/wazuh/core/tests/test_input_validator.py ...                   [ 38%]
framework/wazuh/core/tests/test_logtest.py ...                           [ 38%]
framework/wazuh/core/tests/test_manager.py ............................. [ 39%]
..                                                                       [ 39%]
framework/wazuh/core/tests/test_mitre.py .............                   [ 40%]
framework/wazuh/core/tests/test_pyDaemonModule.py .....                  [ 40%]
framework/wazuh/core/tests/test_results.py ............................. [ 41%]
...........                                                              [ 42%]
framework/wazuh/core/tests/test_rootcheck.py .............               [ 42%]
framework/wazuh/core/tests/test_rule.py .......................          [ 43%]
framework/wazuh/core/tests/test_sca.py .............................     [ 44%]
framework/wazuh/core/tests/test_security.py .............                [ 45%]
framework/wazuh/core/tests/test_stats.py ............................... [ 46%]
.........                                                                [ 47%]
framework/wazuh/core/tests/test_syscheck.py .......                      [ 47%]
framework/wazuh/core/tests/test_syscollector.py ...                      [ 47%]
framework/wazuh/core/tests/test_task.py ........                         [ 47%]
framework/wazuh/core/tests/test_utils.py ............................... [ 49%]
........................................................................ [ 52%]
........................................................................ [ 55%]
........................................................................ [ 58%]
.................................                                        [ 59%]
framework/wazuh/core/tests/test_wazuh_queue.py .......................   [ 60%]
framework/wazuh/core/tests/test_wazuh_socket.py ....................     [ 61%]
framework/wazuh/core/tests/test_wdb.py ...............................   [ 62%]
framework/wazuh/core/tests/test_wlogging.py ............                 [ 63%]
framework/wazuh/rbac/tests/test_auth_context.py ..                       [ 63%]
framework/wazuh/rbac/tests/test_decorators.py .......................... [ 64%]
........................................................................ [ 67%]
...........                                                              [ 67%]
framework/wazuh/rbac/tests/test_default_configuration.py ............... [ 68%]
........................................                                 [ 70%]
framework/wazuh/rbac/tests/test_orm.py ................................. [ 71%]
.............................                                            [ 72%]
framework/wazuh/rbac/tests/test_preprocessor.py ...........              [ 73%]
framework/wazuh/tests/test_active_response.py ............               [ 73%]
framework/wazuh/tests/test_agent.py .................................... [ 75%]
........................................................................ [ 78%]
.....................                                                    [ 79%]
framework/wazuh/tests/test_cdb_list.py ................................. [ 80%]
....................                                                     [ 81%]
framework/wazuh/tests/test_ciscat.py .................................   [ 82%]
framework/wazuh/tests/test_cluster.py ..........                         [ 83%]
framework/wazuh/tests/test_decoder.py .................................. [ 84%]
........................                                                 [ 85%]
framework/wazuh/tests/test_event.py ....                                 [ 85%]
framework/wazuh/tests/test_logtest.py ......                             [ 86%]
framework/wazuh/tests/test_manager.py .................................. [ 87%]
..                                                                       [ 87%]
framework/wazuh/tests/test_mitre.py .......                              [ 87%]
framework/wazuh/tests/test_rootcheck.py ................................ [ 89%]
..................                                                       [ 90%]
framework/wazuh/tests/test_rule.py ..................................... [ 91%]
.....................................                                    [ 93%]
framework/wazuh/tests/test_sca.py ...........                            [ 93%]
framework/wazuh/tests/test_security.py ................................. [ 95%]
......................................                                   [ 96%]
framework/wazuh/tests/test_stats.py ...............                      [ 97%]
framework/wazuh/tests/test_syscheck.py .........................         [ 98%]
framework/wazuh/tests/test_syscollector.py ............                  [ 98%]
framework/wazuh/tests/test_task.py ............................          [100%]

======================= 2373 passed in 271.25s (0:04:31) =======================
```

### CODE REVIEW

File: framework/wazuh/core/cluster/tests/test_cluster.py [hunk 1]
```diff
@@ -603,3 +601,24 @@ def mock_callable(*args, **kwargs):
...
+    config = {cluster.AGENT_CHUNK_SIZE: 120, cluster.AGENT_RECONNECTION_TIME: 10}
+
+    with patch.object(cluster, 'validate') as validate_mock:
+        cluster.validate_haproxy_helper_config(config)
+
+        validate_mock.assert_called_once_with(config, cluster.HAPROXY_HELPER_SCHEMA)
+
+
+def test_validate_haproxy_helper_config_raises_wazuh_error():
+    """Verify that validate_haproxy_helper_config raises WazuhError when the is validation error."""
```
Comment (GGP1) [inline framework/wazuh/core/cluster/tests/test_cluster.py:618]:
```suggestion
    """Verify that validate_haproxy_helper_config raises WazuhError when there is a validation error."""

```

File: framework/wazuh/core/cluster/hap_helper/tests/test_hap_helper.py [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+                    n_managers=len(CLUSTER_NODES.keys()),
+                    server_admin_state_delay=helper.SERVER_ADMIN_STATE_DELAY,
+                )
+
+                if reconnect_agents and len(agent_ids) > 0:
+                    update_agent_connections_mock.assert_called_once_with(agent_list=agent_ids)
+                else:
+                    update_agent_connections_mock.assert_not_called()
+
+    async def test_get_connection_retry_returns_correct_information(
```
[the remark this answers is not in the document]
Comment (nico-stefani) [reply framework/wazuh/core/cluster/hap_helper/tests/test_hap_helper.py:536]: This file is not expected to be modified by the users. So I think is not needed at the moment.

File: framework/wazuh/core/cluster/hap_helper/tests/test_hap_helper.py
[the remark this answers is not in the document]
Comment (nico-stefani) [reply framework/wazuh/core/cluster/hap_helper/tests/test_hap_helper.py:1]: I decided to rename the tests with the proposal.

File: framework/wazuh/core/cluster/hap_helper/tests/test_proxy.py [hunk 3]
```diff
@@ -0,0 +1,833 @@
...
+            params={'version': 0},
+        )
+        assert ret_val == data
+
+    async def test_add_frontend_makes_correct_request(self, proxy_api: ProxyAPI, request_mock: mock.AsyncMock):
+        """Check that `add_frontend` method makes the correct request."""
+
+        endpoint = 'services/haproxy/configuration/frontends'
+        bind_endpoint = 'services/haproxy/configuration/binds'
+        # version_endpoint = 'services/haproxy/configuration/version'
```
[the remark this answers is not in the document]
Comment (nico-stefani) [reply framework/wazuh/core/cluster/hap_helper/tests/test_proxy.py]: Removed

**Review by GGP1 — APPROVED** [review]
LGTM! Great job 🚀

**Review by Selutario — APPROVED** [review]
image

Great job @nico-stefani :clap:

**Pull request merged**